src/gui/qgstextformatwidget.cpp

```diff
@@ -1294,8 +1294,11 @@ void QgsTextFormatWidget::updateFont( const QFont &font )
   // NOTE: QgsFontUtils::fontMatchOnSystem may fail here, just crosscheck family
   mFontMissingLabel->setVisible( !QgsFontUtils::fontFamilyMatchOnSystem( mRefFont.family() ) );
 
-  mDirectSymbLeftLineEdit->setFont( mRefFont );
-  mDirectSymbRightLineEdit->setFont( mRefFont );
+  if ( mDirectSymbolsFrame->isVisible() )
```

Comment:
I'd probably go with this:

   // use actual font family for left/right direction symbol editors, in case user is using a symbol font
   // and we want to show them an accurate representation of the symbols they'll get.
   // While we use the font family, we reset the size to the standard font size used by this widget -- we don't
   // want the line edit fonts to change size to match the label size.
   QFont symbolFont = mRefFont;
   symbolFont.setPointSize( font().pointSize() );
   mDirectSymbLeftLineEdit->setFont( symbolFont );
   mDirectSymbRightLineEdit->setFont( symbolFont );

Reply:
done